training_configs/vilbert_gqa_from_huggingface.jsonnet

```diff
@@ -0,0 +1,92 @@
+local model_name = "bert-base-uncased";
+local effective_batch_size = 128;
+local gpu_batch_size = 32;
+local num_gpus = 1;
+
+local construct_vocab = false;
+
+local vocabulary = if construct_vocab then {
+      // read the files to construct the vocab
+      "min_count": {"answers": 1}
+    } else {
+      // read the constructed vocab
+      "type": "from_files",
+      // CHANGE LOC
+      "directory": "/Users/jacksons/Projects/gqa-train/vocabulary/vilbert_gqa_train.vocab.tar.gz"
+    };
+
+{
+  "dataset_reader": {
+    "type": "gqa",
+    // CHANGE LOC
+    "image_dir": "/Users/jacksons/Projects/gqa-train/images",
+    "feature_cache_dir": "/Users/jacksons/Projects/gqa-train/feature_cache",
+    "image_loader": "detectron",
+    "image_featurizer": "resnet_backbone",
+    "region_detector": "faster_rcnn",
+    "tokenizer": {
+      "type": "pretrained_transformer",
+      "model_name": model_name
+    },
+    "token_indexers": {
+      "tokens": {
+        "type": "pretrained_transformer",
+        "model_name": model_name
+      }
+    },
+    // Change max instances
+    "max_instances": 1000,
+    "image_processing_batch_size": 16,
+    "answer_vocab": if construct_vocab then null else vocabulary,
+    "keep_unanswerable_questions": false
+  },
+  "validation_dataset_reader": self.dataset_reader {
+    "keep_unanswerable_questions": true
+  },
+  "vocabulary": vocabulary,
+  "train_data_path": "testdev_all",
+  "validation_data_path": "val_balanced",
+  "model": {
+    "type": "vqa_vilbert_from_huggingface",
+    "model_name": model_name,
+    "image_feature_dim": 2048,
+    "image_hidden_size": 1024,
+    "image_num_attention_heads": 8,
+    "image_num_hidden_layers": 6,
+    "combined_hidden_size": 1024,
+    "combined_num_attention_heads": 8,
+    "pooled_output_dim": 1024,
+    "image_intermediate_size": 1024,
+    "image_attention_dropout": 0.1,
+    "image_hidden_dropout": 0.1,
+    "image_biattention_id": [0, 1, 2, 3, 4, 5],
+    "text_biattention_id": [6, 7, 8, 9, 10, 11],
+    "text_fixed_layer": 0,
+    "image_fixed_layer": 0,
+    "fusion_method": "mul"
+  },
+  "data_loader": {
+    "batch_size": gpu_batch_size,
+    "shuffle": true,
+    "max_instances_in_memory": 1024
+  },
+  [if num_gpus > 1 then "distributed"]: {
+    #"cuda_devices": std.range(0, num_gpus - 1)
+    "cuda_devices": std.repeat([-1], num_gpus)  # Use this for debugging on CPU
+  },
+  "trainer": {
+    "optimizer": {
+        "type": "huggingface_adamw",
+        "lr": 4e-5
+    },
+    "learning_rate_scheduler": {
+      "type": "linear_with_warmup",
+      "warmup_steps": 300000 / 30,
+      "num_steps_per_epoch": std.ceil(644401 / $["data_loader"]["batch_size"] / $["trainer"]["num_gradient_accumulation_steps"])
```

Comment:
Did you verify that this number makes sense? Is that the correct number in this case?

Is the GQA dataset even large enough that we need to worry about lazy loading? If we don't lazy load, than it figures out this number for us.